### DISCUSSION TITLE
Idea: Gift-cards, codes and vouchers

### DISCUSSION THREAD

**Discussion by bdbelchev** [MartsTech/Aeon#62] [Features] [answered]
This idea consists simply of adding gift code functionality during checkout, which will affect prices.
Codes could be specific to select categories of products, valid only during a certain timeframe, or be valid globally. They could be given to users during their sign-up and first purchase, for inviting a friend, holiday sales etc.

**Comment by MartsTech** [comment, marked as answer]
This should be done with the cart service and a new discout service for example. Seems like a good idea. The discout service will check of certain conditions such as if there are some global events and the user will be notified my email if there is a discout right now. We could use SendGrid to send this emails. The discout service should be able to work in background of the service as a worker service.
